code/weapon/swarm.cpp

```diff
@@ -505,7 +505,7 @@ void turret_swarm_set_up_info(int parent_objnum, ship_subsys *turret, weapon_inf
 	int bank_fired = swp->current_secondary_bank;
 
 	// initialize tsi
-	tsi->weapon_class = WEAPON_INFO_INDEX(wip);
+	tsi->weapon_class = static_cast<int>(wip - Weapon_info.data());
```

Comment:
See previous comment about changing the macro.

Reply:
Ditto my other response.